### PR TITLE
Support kafka streams 3

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4174
Apparently tests didn't pass with kafka-streams 1.x perhaps we should also test some intermediate versions?

### CODE REVIEW

File: instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/TracingIterator.java
Comment (trask) [inline instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/TracingIterator.java:53]: I think there's a similar use case that @mateuszrzeszutek recently ran into recently:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/be090e431a39f54c8d909ac74838ed0a19818322/instrumentation/kafka-streams-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkastreams/StreamTaskInstrumentation.java#L74-L78

would it work to use the same "workaround" for both? (either thread local suppression, or force unwrapping for both)

**Review by trask — COMMENTED** [review]
let's give a day or two for us to wrap up #4065 with @alesj, that PR has been through enough merge conflicts already 😭

File: instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/TracingIterator.java
Comment (laurit) [reply instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/TracingIterator.java:53]: I think I can use thread local for both, I'll just have to move the thread local to a different method that exists in all kafka-streams versions

File: instrumentation/kafka-clients/kafka-clients-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafka/KafkaTracingWrapperUtil.java
Comment (mateuszrzeszutek) [inline instrumentation/kafka-clients/kafka-clients-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafka/KafkaTracingWrapperUtil.java:8]: I wonder if we could rename it to sth like `KafkaClientsConsumerProcessTracing` - since it only affects the CONSUMER `operation = "process"` part of kafka-clients instrumentation

**Review by mateuszrzeszutek — APPROVED** [review]
Let's wait until #4065 is merged though (I think it should be this week)

**Comment by trask** [comment]
@laurit fyi (so we don't duplicate) I'm resolving merge conflicts

**Comment by laurit** [comment]
@trask too late. I think that it might be better to put `KafkaClientsConsumerProcessTracing` into a `bootstrap` module as when kafka-clients and kafka-streams are in different class loaders they would each get their own copy of that class. The same applies to `KafkaConsumerIterableWrapper` and `KafkaConsumerIteratorWrapper` that are used from other instrumentations, probably should have a `KafkaWrapper` interface in bootstrap module.

**Comment by trask** [comment]
No worries, feel free to force push over my changes

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessTracing.java
Comment (trask) [inline instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessTracing.java:8]: can you add a brief comment about why this lives in the bootstrap class loader?

File: instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessWrapper.java
Comment (trask) [inline instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessWrapper.java:8]: same here

**Review by trask — APPROVED** [review]

File: instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessTracing.java
Comment (laurit) [reply instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessTracing.java:12]: added a comment, could you check whether it makes sense

Comment (trask) [reply instrumentation/kafka-clients/kafka-clients-0.11/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/kafka/KafkaClientsConsumerProcessTracing.java:12]: thx 👍

File: gradle-plugins/src/main/kotlin/io/opentelemetry/javaagent/muzzle/MuzzleDirective.kt
Comment (trask) [inline gradle-plugins/src/main/kotlin/io/opentelemetry/javaagent/muzzle/MuzzleDirective.kt:46]: 👍

**Pull request merged**
